restorer_test.go

```diff
@@ -336,39 +385,81 @@ func testRestorerBuilder(buildpackAPI, platformAPI string) func(t *testing.T, wh
 				})
 
 				when("there is a cache=true layer with wrong sha", func() {
+					var wrongSha string
 					it.Before(func() {
-						var meta string
+						wrongSha = "some-made-up-sha"
+						var meta, layerSha string
 						if api.MustParse(buildpackAPI).Compare(api.MustParse("0.6")) < 0 {
 							meta = "cache=true\n"
 						}
-						h.AssertNil(t, writeLayer(layersDir, "buildpack.id", "cache-launch", meta, "some-made-up-sha"))
+						if api.MustParse(platformAPI).Compare(api.MustParse("0.7")) < 0 {
+							layerSha = wrongSha
+						}
+						h.AssertNil(t, writeLayer(layersDir, "buildpack.id", "cache-launch", meta, layerSha))
+
+						appMetaContents := []byte(fmt.Sprintf(`{
+   "buildpacks": [
+       {
+           "key": "buildpack.id",
+           "layers": {
+               "cache-launch": {
+                   "data": {
+                       "cache-launch-key": "cache-launch-val"
+                   },
+                   "cache": true,
+                   "launch": true,
+                   "sha": "%s"
+               }
+           }
+       }
+   ]
+}
+`, wrongSha))
```

Comment:
I find this variable name a bit confusing, as technically the SHA in the app is the "correct" one. Could we just call it `appSHA`?

Reply:
Interesting. I understand what you're saying but I wanted to make it clear that it's a different sha. I took the name from the `it` of this test `there is a cache=true layer with wrong sha`.

Reply:
Maybe `otherSHA`?

Reply:
I changed it to `madeUpSha` because we're setting it to `some-made-up-sha`. What do you think?

Reply:
🤷🏼  I prefer `otherSHA` or `appSHA`, but I won't die on this hill :)

Reply:
😁 No problem, I'll change it to `otherSHA`.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/624/commits/cb53965d3e716161d3a6131385120c3967d1be80.